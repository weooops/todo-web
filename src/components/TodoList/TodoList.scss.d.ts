export const todoList: string;
export const darkorange: string;
export const red: string;
