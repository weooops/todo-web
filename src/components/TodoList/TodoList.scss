.todo-list {
  text-decoration: underline;

  .darkorange {
    color: darkorange;
  }

  .red {
    background-color: $dark-gray;
  }
}