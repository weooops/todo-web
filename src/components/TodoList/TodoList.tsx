import * as React from 'react';

import * as styles from './styles.scss';
import { TodoType } from '../../models';
import Todo from '../Todo';

interface PropsType {
  todos: Array<TodoType>;
}

class TodoList extends React.Component<PropsType> {
  public shouldComponentUpdate(nextProps: PropsType): boolean {
    return nextProps.todos !== this.props.todos;
  }

  public render(): JSX.Element {
    return (
      <div className={styles.todoList}>
        <div className={styles.darkorange}>
          {this._renderTodos()}
        </div>
      </div>
    );
  }

  private _renderTodos(): Array<JSX.Element> {
    const { todos } = this.props;
    return todos.map(todo => (
      <Todo key={todo.id} todo={todo} />
    ));
  }
}

export default TodoList;