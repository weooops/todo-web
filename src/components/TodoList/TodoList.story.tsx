import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { text, withKnobs } from '@storybook/addon-knobs';

import TodoList from './TodoList';

storiesOf('TodoList', module)
  .addDecorator(withKnobs)
  .add('default', () => {
    const props = {
      todos: [
        { id: 1, title: text('title1', '제목1'), message: text('message1', '메시지1'), updatedAt: '2018-01-28 06:52:05' },
        { id: 2, title: text('title2', '제목2'), message: text('message2', '메시지2'), updatedAt: '2018-01-28 06:52:05' },
        { id: 3, title: text('title3', '제목3'), message: text('message3', '메시지3'), updatedAt: '2018-01-28 06:52:05' }
      ]
    };
    return <TodoList {...props} />;
  });
