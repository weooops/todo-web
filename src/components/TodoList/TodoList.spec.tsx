import * as React from 'react';
import { shallow } from 'enzyme';

import TodoList from './TodoList';

it('renders correctly', () => {
  const props = {
    todos: [
      { id: 1, title: '제목1', message: '메시지1', updatedAt: '2018-01-28 06:52:05' },
      { id: 2, title: '제목2', message: '메시지2', updatedAt: '2018-01-28 06:52:05' },
      { id: 3, title: '제목3', message: '메시지3', updatedAt: '2018-01-28 06:52:05' }
    ]
  };
  const wrapper = shallow(<TodoList {...props} />);
  expect(wrapper).toMatchSnapshot();
});
