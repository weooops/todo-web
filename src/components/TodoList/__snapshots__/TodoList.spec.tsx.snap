// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList
    todos={
      Array [
        Object {
          "id": 1,
          "message": "메시지1",
          "title": "제목1",
          "updatedAt": "2018-01-28 06:52:05",
        },
        Object {
          "id": 2,
          "message": "메시지2",
          "title": "제목2",
          "updatedAt": "2018-01-28 06:52:05",
        },
        Object {
          "id": 3,
          "message": "메시지3",
          "title": "제목3",
          "updatedAt": "2018-01-28 06:52:05",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Todo
          todo={
            Object {
              "id": 1,
              "message": "메시지1",
              "title": "제목1",
              "updatedAt": "2018-01-28 06:52:05",
            }
          }
        />,
        <Todo
          todo={
            Object {
              "id": 2,
              "message": "메시지2",
              "title": "제목2",
              "updatedAt": "2018-01-28 06:52:05",
            }
          }
        />,
        <Todo
          todo={
            Object {
              "id": 3,
              "message": "메시지3",
              "title": "제목3",
              "updatedAt": "2018-01-28 06:52:05",
            }
          }
        />,
      ],
      "className": "todoList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "todo": Object {
            "id": 1,
            "message": "메시지1",
            "title": "제목1",
            "updatedAt": "2018-01-28 06:52:05",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "todo": Object {
            "id": 2,
            "message": "메시지2",
            "title": "제목2",
            "updatedAt": "2018-01-28 06:52:05",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "todo": Object {
            "id": 3,
            "message": "메시지3",
            "title": "제목3",
            "updatedAt": "2018-01-28 06:52:05",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Todo
            todo={
              Object {
                "id": 1,
                "message": "메시지1",
                "title": "제목1",
                "updatedAt": "2018-01-28 06:52:05",
              }
            }
          />,
          <Todo
            todo={
              Object {
                "id": 2,
                "message": "메시지2",
                "title": "제목2",
                "updatedAt": "2018-01-28 06:52:05",
              }
            }
          />,
          <Todo
            todo={
              Object {
                "id": 3,
                "message": "메시지3",
                "title": "제목3",
                "updatedAt": "2018-01-28 06:52:05",
              }
            }
          />,
        ],
        "className": "todoList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "todo": Object {
              "id": 1,
              "message": "메시지1",
              "title": "제목1",
              "updatedAt": "2018-01-28 06:52:05",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "todo": Object {
              "id": 2,
              "message": "메시지2",
              "title": "제목2",
              "updatedAt": "2018-01-28 06:52:05",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "todo": Object {
              "id": 3,
              "message": "메시지3",
              "title": "제목3",
              "updatedAt": "2018-01-28 06:52:05",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
